File: TwoPointers/26_Remove_Duplicates_from_Sorted_Array/java/main.java
class Solution {
  public int removeDuplicates(int[] nums) {
      int i=0;
      
      for(int n:nums) {
          // if next value is not same as n, replace it with n.
          if(i == 0 || nums[i-1] < n) {
              nums[i] = n;
              i++;
          }
      }
      return i;
  }
}

File: TwoPointers/26_Remove_Duplicates_from_Sorted_Array/golang/main/main.go
package main

func main() {

}
func removeDuplicates(nums []int) int {
	for i := range nums {
		for j := i + 1; j < len(nums); {
			if nums[i] == nums[j] {
				// if next value j, equals i, which means j is duplicated val, then remove it.
				nums = append(nums[:i], nums[i+1:]...)
				continue
			}
			// if doesn't delete a value in number, increment the j.
			j++
		}
	}

	return len(nums)
}
